server/routerlicious/packages/routerlicious-base/src/alfred/services/deltaService.ts

```diff
@@ -0,0 +1,131 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { toUtf8 } from "@fluidframework/common-utils";
+import { ISequencedDocumentMessage } from "@fluidframework/protocol-definitions";
+import { ITenantManager, MongoManager } from "@fluidframework/server-services-core";
+
+export class DeltaService {
+    constructor(
+        protected readonly mongoManager: MongoManager,
+        protected readonly tenantManager: ITenantManager,
+    ) {
+    }
+
+    public async getDeltas(
+        collectionName: string,
+        tenantId: string,
+        documentId: string,
+        from?: number,
+        to?: number): Promise<ISequencedDocumentMessage[]> {
+        // Create an optional filter to restrict the delta range
+        const query: any = { documentId, tenantId };
+        if (from !== undefined || to !== undefined) {
+            query["operation.sequenceNumber"] = {};
+
+            if (from !== undefined) {
+                query["operation.sequenceNumber"].$gt = from;
+            }
+
+            if (to !== undefined) {
+                query["operation.sequenceNumber"].$lt = to;
+            }
+        }
+
+        const sort = { "operation.sequenceNumber": 1 };
+        return this.queryDeltas(collectionName, query, sort);
+    }
+
+    public async getDeltasFromStorage(
```

Comment:
I am not sure where we use this endpoint so didn't overwrites in my other PR. Do I need to overwrite this?

Reply:
Cool, I will address this in FRS side